File: c/06-io-operations/io_input.c
#include <stdio.h>

int main(void) {
    char name[64];
    int  age;

    printf("Enter your name: ");
    if (fgets(name, sizeof(name), stdin) == NULL) {
        fprintf(stderr, "Error reading name\n");
        return 1;
    }

    /* fgets keeps the newline - replace it with a string terminator */
    for (int i = 0; name[i] != '\0'; i++) {
        if (name[i] == '\n') {
            name[i] = '\0';
            break;
        }
    }

    printf("Enter your age: ");
    if (scanf("%d", &age) != 1) {
        fprintf(stderr, "Error reading age\n");
        return 1;
    }

    printf("Hello, %s! Next year you will be %d.\n", name, age + 1);
    return 0;
}

File: fortran/06-io-operations/io_input.f90
program io_input
    implicit none
    character(len=50) :: name
    integer :: age

    write(*, '(A)') "Enter your name:"
    read(*, '(A)') name

    write(*, '(A)') "Enter your age:"
    read(*, *) age

    write(*, '(A, A, A, I0, A)') "Hello, ", trim(name), &
        "! You are ", age, " years old."
end program io_input
